### PR TITLE
python3Packages.blinkstick: blinkstick: 1.2.0 -> unstable-2023-05-04

### PR DESCRIPTION
**Pull request by perstarkse**
## Description of changes

Update to build from github source instead of pip package. This due to new community patches being merged into main repository. postPatch is no longer needed, and various other updates are merged. No new release version. See [merged pull](https://github.com/arvydas/blinkstick-python/commits/master) requests for more information 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/blinkstick/default.nix [hunk 1]
```diff
@@ -1,35 +1,25 @@
...
-    substituteInPlace setup.py --replace "pyusb==1.0.0" "pyusb>=1.0.0"
-  '';
-
   propagatedBuildInputs = [ pyusb ];
 
-  # Project has no tests
   doCheck = false;
   pythonImportsCheck = [ "blinkstick" ];
 
   meta = with lib; {
```
Comment (eclairevoyant) [inline pkgs/development/python-modules/blinkstick/default.nix:19]: Since you've gone ahead and explicitly prefixed everything with `lib.`:

```suggestion
  meta = {
```

File: pkgs/development/python-modules/blinkstick/default.nix [hunk 2]
```diff
@@ -1,35 +1,25 @@
...
   version = "1.2.0";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "0rdk3i81s6byw23za0bxvkh7sj5l16qxxgc2c53qjg3klc24wcm9";
+  src = fetchFromGitHub {
+    owner = "arvydas";
+    repo = "blinkstick-python";
+    rev = "8140b9fa18a9ff4f0e9df8e70c073f41cb8f1d35";
+    sha256 = "02qfjvbinjid1hp5chi5ms3wpvfkbphnl2rcvdwwz5f87x63pdzm";
```
Comment (eclairevoyant) [inline pkgs/development/python-modules/blinkstick/default.nix:11]: please use SRI hash (meaning, `hash = "something";`)

File: pkgs/development/python-modules/blinkstick/default.nix [hunk 3]
```diff
@@ -1,35 +1,25 @@
...
   pname = "blinkstick";
   version = "1.2.0";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "0rdk3i81s6byw23za0bxvkh7sj5l16qxxgc2c53qjg3klc24wcm9";
+  src = fetchFromGitHub {
+    owner = "arvydas";
+    repo = "blinkstick-python";
+    rev = "8140b9fa18a9ff4f0e9df8e70c073f41cb8f1d35";
```
Comment (eclairevoyant) [inline pkgs/development/python-modules/blinkstick/default.nix:10]: you've picked a random revision which lacks a proper version, so `version` should be updated

File: pkgs/development/python-modules/blinkstick/default.nix [hunk 4]
```diff
@@ -1,35 +1,25 @@
...
 buildPythonPackage rec {
   pname = "blinkstick";
   version = "1.2.0";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "0rdk3i81s6byw23za0bxvkh7sj5l16qxxgc2c53qjg3klc24wcm9";
+  src = fetchFromGitHub {
+    owner = "arvydas";
+    repo = "blinkstick-python";
```
Comment (eclairevoyant) [inline pkgs/development/python-modules/blinkstick/default.nix:9]: any reason for this fork specifically?

File: pkgs/development/python-modules/blinkstick/default.nix [hunk 5]
```diff
@@ -1,35 +1,25 @@
-{ lib
-, buildPythonPackage
-, fetchPypi
-, pyusb
-}:
+{ lib, buildPythonPackage, fetchFromGitHub, pyusb }:
 
 buildPythonPackage rec {
   pname = "blinkstick";
   version = "1.2.0";
```
Comment (eclairevoyant) [inline pkgs/development/python-modules/blinkstick/default.nix:5]: might need to change this to `"unstable-YYYY-MM-dd"` format since there is no proper version

File: pkgs/development/python-modules/blinkstick/default.nix (hunk 4 above)
Comment (eclairevoyant) [reply pkgs/development/python-modules/blinkstick/default.nix:9]: ah, seems like the official author per https://pypi.org/project/BlinkStick/, fair enough.

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/blinkstick/default.nix (hunk 5 above)
Comment (perstarkse) [reply pkgs/development/python-modules/blinkstick/default.nix:5]: many thanks @eclairevoyant for the input.

**Review by eclairevoyant — APPROVED** [review]

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -12992,6 +12992,12 @@
     githubId = 152312;
     name = "Periklis Tsirakidis";
   };
+  perstark = {
+    email = "perstark.se@gmail.com";
+    github = "perstarkse";
+    githubId = 63069986;
+    name = "Per Stark";
+  };
```
Comment (pbsds) [inline maintainers/maintainer-list.nix:13000]: Please move this change to a separate commit titled "maintainers: add perstark", preceeding the changes in blinkstick.

File: pkgs/development/python-modules/blinkstick/default.nix [hunk 7]
```diff
@@ -1,35 +1,25 @@
...
-, buildPythonPackage
-, fetchPypi
-, pyusb
-}:
+{ lib, buildPythonPackage, fetchFromGitHub, pyusb }:
 
 buildPythonPackage rec {
   pname = "blinkstick";
-  version = "1.2.0";
+  version = "unstable-2023-05-04";
```
Comment (pbsds) [inline pkgs/development/python-modules/blinkstick/default.nix:6]:
```suggestion
  format = "setuptools";

```

**Review by pbsds — CHANGES_REQUESTED** [review]
Looking good!

---

Result of `nixpkgs-review pr 246177` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.blinkstick</li>
    <li>python310Packages.blinkstick.dist</li>
    <li>python311Packages.blinkstick</li>
    <li>python311Packages.blinkstick.dist</li>
  </ul>
</details>

---

Please reword the blinkstick commit to `blinkstick: 1.2.0 -> unstable-2023-05-04`, in accordance with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)

**Comment by perstarkse** [comment]
> Looking good!
> 
> Result of `nixpkgs-review pr 246177` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 4 packages built:
> Please reword the blinkstick commit to `blinkstick: 1.2.0 -> unstable-2023-05-04`, in accordance with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)

Thank you for the suggestions! If you notice anything else, please comment again.

**Comment by pbsds** [comment]
LGTM

@ofborg build python310Packages.blinkstick python311Packages.blinkstick

File: pkgs/development/python-modules/blinkstick/default.nix [hunk 8]
```diff
@@ -1,35 +1,26 @@
...
-  # Upstream fix https://github.com/arvydas/blinkstick-python/pull/54
-  # https://github.com/arvydas/blinkstick-python/pull/54/commits/b9bee2cd72f799f1210e5d9e13207f93bbc2d244.patch
-  # has line ending issues after 1.2.0
-  postPatch = ''
-    substituteInPlace setup.py --replace "pyusb==1.0.0" "pyusb>=1.0.0"
-  '';
-
   propagatedBuildInputs = [ pyusb ];
 
-  # Project has no tests
```
Comment (pbsds) [inline pkgs/development/python-modules/blinkstick/default.nix:25]: Please don't remove this comment

**Review by pbsds — APPROVED** [review]

**Comment by mweinelt** [comment]
Can someone nudge the upstream to create a new release? And if we fetch from git tags would also be appreciated.

**Comment by pbsds** [comment]
https://github.com/arvydas/blinkstick-python/issues/89
